ddtrace/internal/datadog/profiling/dd_wrapper/src/sample.cpp

```diff
@@ -316,6 +318,35 @@ Datadog::Sample::push_class_name(std::string_view class_name)
     return true;
 }
 
+bool
+Datadog::Sample::push_monotonic_ns(int64_t _monotonic_ns)
```

Comment:
Could someone provide me some context why functions here are named `push_*` instead of `add_*` or something else?

Reply:
There's no _great_ reason, but I chose "push" instead of "add" because

1. Data transfers into libdatadog is mediated through an intermediate Sample object; we haven't committed the data until that Sample gets flushed.  As much as we "add" to the Sample, I was hoping that, on the Python side, there's this understanding that data isn't really saved until the Sample has been flushed.
2. State crosses an FFI boundary--we seldom, if ever, manipulate storage directly.

Happy to talk about this.  It's not set in stone.